examples/platforms/utils/settings_flash.c

```diff
@@ -198,10 +197,20 @@ static uint32_t swapSettingsBlock(otInstance *aInstance)
     setSettingsFlag(sSettingsBaseAddress, (uint32_t)OT_SETTINGS_IN_USE);
     setSettingsFlag(oldBase, (uint32_t)OT_SETTINGS_NOT_USED);
 
-exit:
     return settingsSize - sSettingsUsedSize;
 }
 
+#else // #if (SETTINGS_CONFIG_PAGE_NUM > 1)
```

Comment:
I'm not sure `SETTING_CONFIG_PAGE_NUM = 0` is a valid configuration. May be better to have a preprocessor error condition in this case.

Reply:
Do you mean we add a case `#elif (SETTINGS_CONFIG_PAGE_NUM == 1)` here and then error on `#else` covering the case of it being zero?
I think we need to cover `SETTINGS_CONFIG_PAGE_NUM == 1` differently.